lib/ocpp/v201/database_handler.cpp

```diff
@@ -227,6 +236,41 @@ void DatabaseHandler::insert_availability(int32_t evse_id, int32_t connector_id,
     }
 }
 
+void DatabaseHandler::process_interrupted_transactions() {
```

Comment:
I think it would be better to replace this function with a `get_transaction` function with an `evse_id` argument that then returns an optional result. Maybe you could even return an enhanced transaction type so we don't need the `InterruptedTransaction` type anymore.

This could simplify the continuation of transaction in the Evse a bit. It also removes the need to remember these transactions in the database handlers memory like we have it now.